src/controllers/ConfirmationController.ts

```diff
@@ -31,23 +30,13 @@ export class ConfirmationController extends SafeNavigationBaseController<any> {
     }
 
     protected prepareViewModelFromSession(session: Session): Record<string, any> {
-        const companyNumber = session
-            .getExtraData()
-            .chainNullable<ApplicationData>(data => data[APPLICATION_DATA_KEY])
-            .chainNullable(applicationData => applicationData.appeal.penaltyIdentifier)
-            .map(penaltyIdentifier => penaltyIdentifier.companyNumber)
-            .extract();
-
-        const userEmail = session
-            .getValue<ISignInInfo>(SessionKey.SignInInfo)
+        const userProfile = session.getValue<ISignInInfo>(SessionKey.SignInInfo)
             .map(info => info[SignInInfoKeys.UserProfile])
-            .map(userProfile => userProfile?.email)
-            .extract();
+            .orDefault({});
```

Comment:
Behave more expressive there. If user profile is not there blow up. A `unsafeCoerce` will do the job.